rfc/rfc-81/rfc-81.md

```diff
@@ -0,0 +1,157 @@
+
+### Delta Commit
+When `compaction_with_merge_sort_enable` is true, DeltaCommit sorts the written data by RecordKey to achieve orderly records in LogBlock, and sets the LogBlock header's IS_ORDERED to true.
+
+### Compaction
+Iterate over the record iterator returned by the merge-sort based scanner and write to the base file.
+
+## Rollout/Adoption Plan
+- What impact (if any) will there be on existing users?
+  None
+- If we are changing behavior how will we phase out the older behavior?
+  None
+- If we need special migration tools, describe them here.
+  None
+- When will we remove the existing behavior.
+  None
+
+## Performance Test Results
+### Compaction
+Flink MOR table, 7 VCore, 36 GB memory, flink checkpoint 10min, other configurations remain default.
+
+| Type                        | Kafka Fully Consumed | Runtime | Records Consumed | Record Size   |
+|-----------------------------|----------------------|---------|------------------|---------------|
+| Map-based compaction        | No                   | 11.5h   | 226,250,649      | 752 GB        |
+| Merge-sort based compaction | No                   | 11.5h   | 260,188,246      | 858 GB        |
+
+### Log Compaction
+Flink MOR table, 7 VCore, 36 GB memory, flink checkpoint 10min, other configurations remain default.
+
+| Type                            | Kafka Fully Consumed | Runtime | Records Consumed | Record Size   |
+|---------------------------------|----------------------|---------|------------------|---------------|
+| Map-based log compaction        | No                   | 11.5h   | 147,995,817      | 494 GB        |
+| Merge-sort based log compaction | No                   | 11.5h   | 170,726,805      | 573 GB        |
+
+Noted:
+1. Delta Commit + Log Compaction: This approach has no advantage. Log Compaction is slower than Compaction because the log data grows significantly, leading to increased read times for log files. 
+   Log Compaction operates in append mode, which increases the log size with each execution. Additionally, AVRO files are five times larger than Parquet files in our test table.
+2. Delta Commit + Compaction + Log Compaction: Log Compaction to merge Log Block fragments in order to improve the query speed for Readers, while test results show no significant impact on write performance consumption.
+   This RFC does not include Log Compaction, because it primarily supports the MDT primary key index, so it is recommended to include it in the new RFC for the MDT primary key index.
+   Reference [Balancing compute cost and query performance](https://docs.google.com/presentation/d/10hHkQsd0bCdCor4w9Wduds5l-i_Z7X7A-zk_vYYi4kA/edit#slide=id.p18)
+
+### Sparse Index for MDT Primary Key
+This RFC does not involve building a primary key index for MDT, but we provide query performance test results for the primary key index:
+
```

Comment:
What is a `primary key index`, can we elaborate a little more?

Reply:
> What is a `primary key index`, can we elaborate a little more?

@danny0405 done